### DISCUSSION TITLE
Stripe Payment

### DISCUSSION THREAD

**Discussion by silicahd** [givanz/Vvveb#64] [Ideas]
Stripe is the most used payment. At the moment I do not know enough about this system to implement it but would be nice to have as a standard for others.

**Comment by givanz** [comment]
A payment gateway plugin is in development using omnipay library, it will have support for most major providers https://github.com/thephpleague/omnipay?tab=readme-ov-file#payment-gateways like Stripe and Paypal.

Once it's stable it will be added to plugins market for public testing.

**Comment by 963design** [comment]
I been testing in development both plugins, stripe and omnipay wich one is recommended or more stable for production?

**Reply by givanz** [reply]
It is better to use the native stripe plugin as it uses the latest version of the official stripe php library.

The omnipay library is not actively maintained and is recommended to be used for paypal or for code familiarity with the omnipay library for extending or customizing the plugin.

Let me know if you encounter any issues with the plugins.

**Reply by 963design** [reply]
thanks, actually we have been testing in a vvveb instance for a client's ecommerce, but we have encountered several issues, and some weird behavior (some times shows the payment and shipping options and some times don't in the checkout page) , we'll like to know if is something wrong with our setup our configuration, we are running on a linux server with php 8.2 with all the suggested extensions and we clone the default template to avoid loosing functionality but maybe we broke something, what else we can try to identify the problem?

**Comment by Ezyweb-uk** [comment]
Is the Stripe plugin working with Vvveb version 1.0.7? I tried downloading it from https://plugins.vvveb.com/product/stripe and uploading it, but I got the error: Method does not exist!

**Comment by givanz** [comment]
Can you please provide more details? 

The method does not exist error is shown on checkout page before placing the order or after the order is placed or it's shown somewhere else?

I just tested Stripe with 1.0.7 and I can't reproduce this error, tested with docker image.

You can install the plugin from the dashboard without downloading it first.
 
[stripe.webm](https://github.com/user-attachments/assets/7a41f178-4308-4267-8f30-3440a660e099)

**Comment by Ezyweb-uk** [comment]
The installation worked OK via the '+ Install Plugin' button, @givanz thanks for your help!

The previous error had occurred after downloading/uploading the plugin from https://plugins.vvveb.com/product/stripe, I think it was immediately after uploading or activating, before loading other pages.

Does the Stripe plugin have a repo page, as the link at the bottom of https://plugins.vvveb.com/product/stripe to  https://github.com/Vvveb/stripe gives a 404.

Also the Omnipay plugin mentions Stripe but it's not listed in the Gateway options.

**Reply by givanz** [reply]
Stripe plugin is not pushed to github yet, I will add all plugins to their own repository, now only a handful are on github.

You can view all the plugin code inside plugins/stripe folder.

Omnipay plugin does not have the stripe payment intent frontend integration yet, it only works with paypal for now and it's provided for easy integration with the available omnipay providers.
